src/modules/lua_mtev.c

```diff
@@ -953,9 +961,38 @@ static int
 mtev_lua_socket_sock_name(lua_State *L) {
   return mtev_lua_socket_gen_name(L, getsockname);
 }
+
+static int
+socket_connect_timeout(eventer_t e, int mask, void *closure, struct timeval *now) {
+  (void)mask;
+  (void)now;
+  mtev_lua_resume_info_t *ci;
+  lua_timeout_callback_ref* cb_ref;
+  lua_State *L;
+
+  cb_ref = (lua_timeout_callback_ref*) closure;
+  L = cb_ref->L;
+  ci = mtev_lua_find_resume_info(L, mtev_true);
+  assert(ci);
+
+  // remove the original event
+  eventer_remove_fde(cb_ref->timed_out_eventer);
+  mtev_lua_deregister_event(ci, cb_ref->timed_out_eventer, 0);
+
```

Comment:
There are two additional lines in the socket_read_timeout function, that I don't understand, and were thus not copied over to our timeout handler:
```
*(cl->eptr) = eventer_alloc_copy(cb_ref->timed_out_eventer);
mtev_lua_register_event(ci, *cl->eptr);
```
For me it looks like we are re-scheduling the timeout event. Which makes no sense to me.

Reply:
We have to track the event, otherwise it will leak.. It is no longer tracked in the eventer (eventer_remove_fde)... so  e will be freed when we return and we can no longer keep a reference to it.  So we make a copy of it and put that in our lua state (mtev_lua_register_event).  This way, when the coro finishes, we know that we can close that event out (fd).

Reply:
Where does the "timed_out_eventer" get closed?

Reply:
I see. I'll change the code to make a copy of the event and update cl->eptr.

> Where does the "timed_out_eventer" get closed?
It will get closed when we GC the original eventer object (the one created by mtev.socket()).
It might be better to close it here, though.

Reply:
opted against the automatic close. With C/POSIX sockets you also have to call close explicitly.